src/executionTimingsReporter.ts

```diff
@@ -0,0 +1,128 @@
+import { HighResTimer, calculateElapsedTime } from "./utils";
+import { EventEmitter } from "events";
+
+export type TimingsReport = {[stepName: string]: [number] | number};
+
+const stepStartedEventName = 'stepStarted';
+const milestoneReachedEventName = 'milestoneReached';
+
+interface StepStartedEventArguments {
+    stepName: string;
+}
+
+interface MilestoneReachedEventArguments {
+    milestoneName: string;
+}
+
+export interface ObservableEvents {
+    Events: EventEmitter;
+}
+
+export class StepStartedEventsEmitter extends EventEmitter {
+    constructor(public readonly NestedEmitters: [EventEmitter] = [] as [EventEmitter]) {
+        super();
+    }
+
+    public emitStepStarted(stepName: string): void {
+        this.emit(stepStartedEventName, { stepName: stepName } as StepStartedEventArguments);
+    }
+
+    public emitMilestoneReached(milestoneName: string): void {
```

Comment:
Should we make a correspondence between two method? One is `start`, the other is `finish/compete`? instead of introducing a new concept of `milestone` (which in some sense might collide with the concept of `step` )

Reply:
And who calls this now?

Reply:
This gets called from vscode-chrome-debug in the other PR. Let's sync offline about your concern...